File: data/code_pairs/fortran/918.f90
module ModExample2

  implicit none
  private
  public :: example_iroutine
  character(len=*), parameter :: NameMod = 'ModExample2'

contains

  subroutine example_iroutine(InputVar, OutputVar, InputOutputVar)

    ! Assuming ExampleParameter is a real number for demonstration purposes
    real, parameter :: ExampleParameter = 1.0

    real, intent(in) :: InputVar
    real, intent(out) :: OutputVar
    real, intent(inout) :: InputOutputVar

    character(len=*), parameter :: NameSub = NameMod//'::example_iroutine'

    write(*,*) NameSub, ': documented executable statements come here'
    write(*,*) NameSub, ': this part should not appear in the documentation'

    ! Example computation to demonstrate functionality
    OutputVar = InputVar * ExampleParameter * 2.0
    InputOutputVar = InputOutputVar * 3.0

  end subroutine example_iroutine

end module ModExample2

File: data/code_pairs/cpp/918.cpp
#ifndef MODEXAMPLE2_H
#define MODEXAMPLE2_H

#include <iostream>
#include <string>

namespace ModExample2 {

    constexpr char NameMod[] = "ModExample2";
    const float ExampleParameter = 1.0f; // Assuming ExampleParameter is defined here for simplicity

    void example_iroutine(float InputVar, float& OutputVar, float& InputOutputVar);

} // namespace ModExample2

#endif // MODEXAMPLE2_H